Celeste.Mod.mm/Patches/StrawberrySeed.cs

```diff
@@ -38,3 +47,42 @@ public patch_StrawberrySeed(Strawberry strawberry, Vector2 position, int index,
         }
     }
 }
+namespace MonoMod {
+    /// <summary>
+    /// Patches OnAllCollected to add a check if this.follower.Leader is null.
+    /// </summary>
+    [MonoModCustomMethodAttribute(nameof(MonoModRules.PatchStrawberrySeedOnAllCollected))]
+    class PatchPatchStrawberrySeedOnAllCollectedAttribute : Attribute { }
+
+    static partial class MonoModRules {
+
+        public static void PatchStrawberrySeedOnAllCollected(ILContext context, CustomAttribute attrib) {
+            ILCursor cursor = new(context);
+            ILLabel beforeLoseFollower = cursor.DefineLabel();
+            ILLabel afterLoseFollower = cursor.DefineLabel();
+
+            // we want to add a null check on Leader before calling LoseFollower so we change:
+            // this.follower.Leader.LoseFollower(this.follower);
+            // to
+            // this.follower.Leader?.LoseFollower(this.follower);
+
+            // move cursor to the point where Leader is on top of the stack, and then duplicate it for the branch true
+            cursor.GotoNext(MoveType.After, inst => inst.MatchLdfld("Celeste.Follower", "Leader"));
+            cursor.Emit(OpCodes.Dup);
+            
+            // brach to the calling path if we are not null
+            cursor.Emit(OpCodes.Brtrue, beforeLoseFollower);
+            // on null discard duplicated value and jump over the calling path
+            cursor.Emit(OpCodes.Pop);
+            cursor.Emit(OpCodes.Br, afterLoseFollower);
+            
+            cursor.GotoNext(MoveType.Before, inst => inst.MatchLdarg(0));
+            cursor.MarkLabel(beforeLoseFollower);
+            
+            cursor.GotoNext(MoveType.After, inst => inst.MatchCallvirt("Celeste.Leader", "LoseFollower"));
+            cursor.GotoNext();
```

Comment:
this `cursor.GotoNext();` seems unnecessary

Reply:
You are right, it is